Lib/typing.py

```diff
@@ -1216,7 +1216,9 @@ def _determine_new_args(self, args):
         return tuple(new_args)
 
     def copy_with(self, args):
-        return self.__class__(self.__origin__, args, name=self._name, inst=self._inst)
+        return self.__class__(self.__origin__, args, name=self._name, inst=self._inst,
+                              _typevar_types=self._typevar_types,
+                              _paramspec_tvars=self._paramspec_tvars)
```

Comment:
This (perplexingly) no longer works after merging #27518.

Reply:
Do your tests pass with https://github.com/python/cpython/pull/27518 but without modifying this line of code? I have a suspicion that PR accidentally fixed our bug too.

If that's the case, we just need to merge the tests here.

Reply:
Sorry for being vague. #27518 breaks this PR.

After #27518, and **_with_** the change highlighted by this comment thread, these tests now fail:

ERROR: test_concatenate (test.test_typing.CollectionsCallableTests)
ERROR: test_var_substitution (test.test_typing.ConcatenateTests)
ERROR: test_concatenate (test.test_typing.TypingCallableTests)

After #27518, but **_without_** this change, the test introduced elsewhere in this PR fails:

FAIL: test_copy_with (test.test_typing.GenericTests)

So #27518 breaks my approach for `_GenericAlias.copy_with`, but the problem identified in [bpo-46581](https://bugs.python.org/issue46581) still remains. I'm still trying to grok why, but `typing.py` internals are very hard to follow, with a lot of non-intuitive interactions and layers that seem to grow as time goes by, not less.

I may be naive, but, intuitively, the change herein _should_ make sense. #26091 added members to the class and exposed setting them via the `__init__` method. The `copy_with` method already invoked the `__init__` method to make a copy. This change propagates the new members via the `__init__` method, which arguably should have been done as part of #26091.

#27518 seems to add even more layers of indirection/complexity. It's not even obvious how those changes interact with these. Perhaps it just isn't for me to understand. Maybe there's genius at work here, but some overarching design (if it exists) is hard for me to tease out. To me, it just appears as a nearly unnavigable hodgepodge of intertwined kludges. 🤷

I think at this point, this will likely require someone more familiar with that code base. My inclination is to close this PR as a failed attempt and leave [bpo-46581](https://bugs.python.org/issue46581) open for someone else to fix, since I'm unclear how to proceed in light of #27518.